### ISSUE TITLE
StagingBelt memory usage problems in certain scenarios

### ISSUE THREAD

**Issue by eduardobraun** [gfx-rs/wgpu#1441]
Recently I decided to try out Iced but run into some memory problems while testing an example, as can be seen in this issue: https://github.com/hecrj/iced/issues/786
I narrowed it down to the wgpu-rs StagingBelt, and the lack of a way to cleanup unused free Chunks from the belt.
The issue with the Game of Life example was that I only added more geometry to the screen, so my buffers got larger, never fitting inside the previous Chunks and so creating new Chunks while the old Chunks stayed unused forever.
I managed to fix the issue by implementing a naive GC: https://github.com/eduardobraun/wgpu-rs/commit/a62dd38538b41e2890a0cecd59e692f1d208d6e6

TLDR: StagingBelt needs a way to reclaim memory from free Chunks.
